storybook/cypress/e2e/header.cy.ts

```diff
@@ -70,104 +71,113 @@ describe("Header", () => {
+      .get("@dsoHeaderShadow")
+      .find("nav")
       .should("exist")
       .and("be.visible")
       .find("ul li dso-dropdown-menu")
       .should("not.exist")
       .and("be.null")
       .viewport(992, 600)
-      .get("dso-header")
+      .get("@dsoHeaderShadow")
       .find("nav ul li dso-dropdown-menu")
       .should("exist")
       .and("be.visible")
       .viewport(568, 600)
-      .get("dso-header")
+      .get("@dsoHeaderShadow")
       .find("nav ul li dso-dropdown-menu")
       .should("not.exist")
       .and("be.null");
   });
 
   it("should act on user-profile attributes", () => {
-    cy.get("dso-header")
+    cy.get("dso-header.hydrated")
       .invoke("attr", "auth-status", "loggedIn")
+      .get("@dsoHeaderShadow")
       .find(".profile")
       .should("be.visible")
       .get("dso-header")
       .invoke("removeAttr", "user-profile-url")
       .invoke("removeAttr", "user-profile-name")
+      .get("@dsoHeaderShadow")
       .find(".profile")
       .should("not.exist");
   });
 
   it("should act on user-home-url attribute", () => {
-    cy.get("dso-header")
+    cy.get("dso-header.hydrated")
       .invoke("attr", "auth-status", "loggedIn")
+      .get("@dsoHeaderShadow")
       .find("nav li.menu-user-home")
       .should("be.visible")
       .get("dso-header")
       .invoke("removeAttr", "user-home-url")
-      .get("dso-header")
+      .get("@dsoHeaderShadow")
       .find("nav li.menu-user-home")
       .should("not.exist");
   });
 
   it("should act on show-help attribute", () => {
-    cy.get("dso-header")
+    cy.get("dso-header.hydrated")
       .invoke("attr", "show-help", true)
+      .get("@dsoHeaderShadow")
       .find(".dso-header-session .help button")
       .should("be.visible");
   });
 
   it("should use an anchor if help-url is passed", () => {
-    cy.get("dso-header")
+    cy.get("dso-header.hydrated")
       .invoke("attr", "show-help", true)
       .invoke("attr", "help-url", "#help")
+      .get("@dsoHeaderShadow")
       .find(".dso-header-session .help a")
       .should("be.visible");
   });
 
   it("should not show menu", () => {
-    cy.visit("http://localhost:45000/iframe.html?id=core-header--with-label&args=noMainMenu:true")
-      .get("dso-header")
+    cy.visit("http://localhost:45000/iframe.html?id=core-header--with-label&args=noMainMenu:true");
+
+    prepareComponent();
+
+    cy.get("dso-header.hydrated")
       .invoke("removeAttr", "user-home-url")
+      .get("@dsoHeaderShadow")
       .find("nav")
       .should("not.exist")
-      .get("dso-header")
+      .get("@dsoHeaderShadow")
       .find("dso-dropdown-menu")
       .should("not.exist");
   });
 
   it("should show login or logout when no menuItems are provided", () => {
-    prepareComponent();
-
-    cy.get("dso-header")
-      .then(($header) => setMenuItems($header, []))
+    cy.get("dso-header.hydrated")
+      .then(($header: JQuery<HTMLDsoHeaderElement>) => setMenuItems($header, []))
```

Comment:
Volgens mij kan dit ook:

```suggestion
    cy.get<HTMLDsoHeaderElement>("dso-header.hydrated")
      .then(($header) => setMenuItems($header, []))
```

Het fraaie is dat de type (`<HTMLDsoHeaderElement>`) op bij de selector (`"dso-header.hydrated"`) staat, waardoor de code beter refactorbaar blijft. Bijvoorbeeld als de `.get()` en `.then()` worden gescheiden.

_Ter lering en vermaak:_

```
    cy.get("dso-header")
```

Dit begrijpt Cypress en zal de juiste element interface in de chain tonen.

Maar dat is ontoereikend, want wij willen past testen als de class `hydrated` is toegevoegd. Een alternatief is:

```
    cy.get("dso-header")
      .should("have.class", "hydrated")
```

Dit zorgt ervoor dat Cypress de juiste element interface plaatst, pas verder gaat als er een class `hydrated` op zit, en uit `.should()` komt hetzelfde subject als wat er in ging.

Reply:
Heb je suggestie overgenomen